deploy/charts/monitoring/templates/deployment.yaml

```diff
@@ -53,94 +53,60 @@ spec:
       {{- end}}
       serviceAccount: monitoring
       containers:
-      - name: influxdb
-        image: {{ required "An image is required" .Values.image }}
-        lifecycle:
-          postStart:
-            exec:
-              command: ["/bin/sh", "-c", "/influx-post-start.sh"]
-        command:
-        - /influxd
-        resources:
-          {{- toYaml .Values.resources | nindent 10 }}
-        securityContext:
-          capabilities:
-            drop: ["NET_RAW"]
-          readOnlyRootFilesystem: true
-        volumeMounts:
-          - name: influxdb-config-volume
-            mountPath: /etc/influxdb/
-            readOnly: true
-          - name: data-volume
-            mountPath: /var/lib/influxdb
-      - name: telegraf-proxy
-        image: {{ required "An image is required" .Values.image }}
-        command:
-        - /telegraf
-        ports:
-        - name: api
-          containerPort: 443
-        resources:
-          requests:
-            memory: "100Mi"
-            cpu: "100m"
-          limits:
-            memory: "200Mi"
-            cpu: "200m"
+      - name: grafana
+        image: {{ required "A Grafana image is required" .Values.grafanaImage }}
         securityContext:
           capabilities:
             drop: ["NET_RAW"]
           readOnlyRootFilesystem: true
-        volumeMounts:
-          - name: monitoring-db-volume
-            mountPath: /run/secrets/stackrox.io/monitoring/certs/
-            readOnly: true
-          - name: telegraf-proxy-config-volume
-            mountPath: /etc/telegraf
-            readOnly: true
-      - name: grafana
-        image: {{ required "An image is required" .Values.image }}
-        command:
-        - /grafana-entrypoint.sh
-        - /run.sh
-        ports:
-        - name: api
-          containerPort: 3000
         resources:
           requests:
             memory: "100Mi"
             cpu: "100m"
           limits:
             memory: "500Mi"
             cpu: "500m"
-        securityContext:
-          capabilities:
-            drop: ["NET_RAW"]
-          readOnlyRootFilesystem: true
         volumeMounts:
           - name: grafana-config-volume
             mountPath: /etc/grafana
+            readOnly: true
```

Comment:
nit: does this mean you can't change grafana config afterwards within the UI due to it now being read only? We can still do it via cm change, just curious.

Reply:
So `/etc/grafana` is used by Grafana for static configuration - e.g. provisioning of dashboards and such. Therefore no write access is needed. Changes to the UI are still possible and would be stored in `/var/lib/grafana`. However, you are still right that such changes would not be persisted, because it is mounted without PVC.

The idea is to provision the dashboard declaratively - on every restart the dashboard should be the same. This is also in line with how the dashboards in the managed service world will work. To add a new dashboard, one can create the dashboard the UI, then export the JSON and add it to the provisioning.